### PR TITLE
[Release] NNTrainer v0.5.0 release 

### PR DESCRIPTION
**Pull request by jijoongmoon**
NNTrainer v0.5.0 is released.

This PR includes rebase as well.

**Changes proposed in this PR:**
- Added TOC generator for README.md

Resolves:

**Self evaluation:**
1. Build test:	 [X]Passed [ ]Failed [ ]Skipped
2. Run test:	 [X]Passed [ ]Failed [ ]Skipped

Signed-off-by: jijoong.moon <jijoong.moon@samsung.com>

### CODE REVIEW

**Review by lhs8928 — APPROVED** [review]
LGTM

**Review by DonghakPark — APPROVED** [review]
LGTM!

File: Applications/Custom/meson.build [hunk 1]
```diff
@@ -2,8 +2,11 @@
 layer_example_pow_src = files('pow.cpp')
 layer_example_mae_src = files('mae_loss.cpp')
 layer_example_rnnt_src = files('rnnt_loss.cpp')
+optimizer_example_momentum_src = files('momentum.cpp')
 
 layer_example_inc = include_directories('./')
+optimizer_example_inc = include_directories('./')
```
Comment (SeoHyungjun) [inline Applications/Custom/meson.build:8]: Aren't layer_example_inc and optimizer_example_inc the same path?

**Review by SeoHyungjun — APPROVED** [review]
LGTM!

**Pull request merged**